### PR TITLE
Ast docs

### PR DESCRIPTION
**Pull request by arichiardi**
This patch officially include `replumb.ast` to the public functions (with codox doc of course)

### CODE REVIEW

File: src/cljs/replumb/ast.cljs [hunk 1]
```diff
@@ -12,10 +13,10 @@
   "Given compiler state and namespace symbol return all the public vars
   in the AST.
 
-  Analagous to cljs.analyzer/ns-publics, returns var analysis maps not
-  vars.
+  Analagous to `cljs.analyzer/ns-publics`, but it returns analysis maps
```
Comment (tomasz-biernacki) [inline src/cljs/replumb/ast.cljs]: Analogous

File: src/cljs/replumb/ast.cljs [hunk 2]
```diff
@@ -28,10 +29,10 @@
   "Given compiler state and namespace symbol return all the vars in the
   AST.
 
-  Analagous to cljs.analyzer/ns-interns, returns var analysis maps not
-  vars.
+  Analagous to `cljs.analyzer/ns-interns`, but it returns analysis maps
```
Comment (tomasz-biernacki) [inline src/cljs/replumb/ast.cljs]: Analogous

File: src/cljs/replumb/ast.cljs [hunk 3]
```diff
@@ -39,68 +40,75 @@
    (get-in state [:cljs.analyzer/namespaces ns :defs])))
 
 (defn ns-defs
-  "Given compiler state and namespace symbol, returns its AST :defs
-  content (beware, it can be a lot of data)"
+  "Given compiler state and namespace symbol, returns its AST's `:defs`
```
Comment (tomasz-biernacki) [inline src/cljs/replumb/ast.cljs]: _the_ compiler state

File: src/cljs/replumb/ast.cljs [hunk 4]
```diff
@@ -39,68 +40,75 @@
...
+
+  Beware, there can be a lot of data."
   [state ns]
   {:pre [(symbol? ns)]}
   (get-in state [:cljs.analyzer/namespaces ns]))
 
 (defn requires
-  "Return required symbols given compiler state and a namespace: a
-  map of {ns ns}.
+  "Return required symbols given compiler state and namespace: a map of
```
Comment (tomasz-biernacki) [inline src/cljs/replumb/ast.cljs]: Return_s_

File: src/cljs/replumb/ast.cljs (hunk 3 above)
Comment (arichiardi) [inline src/cljs/replumb/ast.cljs]: I opted to remove articles everywhere here

File: src/cljs/replumb/ast.cljs [hunk 5]
```diff
@@ -39,68 +40,75 @@
...
+  You need a `require` in the requirer-ns namespace for this to return
+  something."
   [state requirer-ns]
   {:pre [(symbol? requirer-ns)]}
   (get-in state [:cljs.analyzer/namespaces requirer-ns :requires]))
 
 (defn imports
-  "Return imported symbols given compiler state and a namespace: a
-  map of {symbol ns}.
+  "Return imported symbols given compiler state and a namespace: a map
```
Comment (tomasz-biernacki) [inline src/cljs/replumb/ast.cljs]: Return*s*

File: src/cljs/replumb/ast.cljs (hunk 4 above)
Comment (arichiardi) [inline src/cljs/replumb/ast.cljs]: I opted to remove `s` in the first sentence only, I see it somewhere, they use imperative (?) tense

File: src/cljs/replumb/ast.cljs [hunk 6]
```diff
@@ -200,14 +218,17 @@
 (defn dissoc-all
   "Dissoc all the required-ns symbols from requirer-ns.
 
-  There are three types of symbol in replumb jargon, which loosely map
-  to cljs.js ones. These optionally go in the type parameter as keyword:
+  There are three types of symbol in replumb jargon, which loosely map to
```
Comment (tomasz-biernacki) [inline src/cljs/replumb/ast.cljs]: three types of symbol_s_ (maybe)

File: src/cljs/replumb/ast.cljs [hunk 7]
```diff
@@ -200,14 +218,17 @@
...
+  `cljs.js` ones. These optionally go in the type parameter as keyword:
 
-  - :symbol, for instance my-sym in (def  my-sym 3), the default
-  - :macro, which comes from a defmacro
-  - :import, for instance User in (import 'foo.bar.User)
-  - :require, which is the namespace symbol in a (require ...)
-  - :macro-require, which is the namespace symbol in a (require-macros ...)"
+  - `:symbol`, the default, for instance my-sym in `(def  my-sym 3)`
```
Comment (tomasz-biernacki) [inline src/cljs/replumb/ast.cljs]: Extra space in `(def  my-sym 3)`

Comment (arichiardi) [inline src/cljs/replumb/ast.cljs]: :+1:

File: src/cljs/replumb/ast.cljs (hunk 6 above)
Comment (arichiardi) [inline src/cljs/replumb/ast.cljs]: I think no here, the plural is on type.

**Pull request merged**